python/triton/compiler/code_generator.py

```diff
@@ -208,15 +208,16 @@ def visit_Call(self, node: ast.Call) -> bool:
 class CodeGenerator(ast.NodeVisitor):
 
     def __init__(self, context, prototype, gscope, attributes, constants, function_name, jit_fn: JITFunction, options,
-                 debug=None, module=None, is_kernel=False, function_types: Optional[Dict] = None, noinline=False,
-                 file_name: Optional[str] = None, begin_line=0):
+                 codegen_fn, debug=None, module=None, is_kernel=False, function_types: Optional[Dict] = None,
```

Comment:
I am confused by the codegen_fn variable.  I don't see it documented anywhere, and also it seems to be a dict of functions, not a fn itself?  The functions in the dict have an implicit interface that I think we also need to write down.

Reply:
renamed it to `codegen_fns` and added a comment in `CodeGenerator`. I'm not sure where you would want to write down the signature implied by the function.